src/lib.rs

```diff
@@ -987,6 +999,23 @@ impl<A: Array> Clone for ArrayVec<A>
     }
 }
 
+#[cfg(feature="quickcheck-1")]
+impl<A: Array + Send + Clone + 'static > Arbitrary for ArrayVec<A>
+where <A as array::Array>::Item: std::clone::Clone + Arbitrary,
+<A as array::Array>::Index: std::marker::Send
+{
+    fn arbitrary<G: Gen>(g: &mut G) -> Self {
+        use std::cmp::{min, max};
+        let mut v = Self::default();
+        let size = g.gen_range(0, max(1, min(g.size(), v.capacity())));
+        (0..size).for_each(
```

Comment:
This is probably more idiomatically implemented with a for loop.

Reply:
Fixed!